### PR TITLE
feat: ollama integration

### PR DESCRIPTION
**Pull request by arjansingh**
1. add support for ollama's built in API
2. adds a rendered output option so tools can pass around output rendered from our render pipeline directly to each other during execution scripts.

### CODE REVIEW

File: script/engine.go [hunk 1]
```diff
@@ -242,6 +251,65 @@ func (e *Engine) Run(ctx context.Context, source string) (*mcp.ToolResult, error
...
+		return nil, fmt.Errorf("failed to set api.callRendered: %w", err)
+	}
+
+	if err := apiObj.Set("tryCallRendered", func(call goja.FunctionCall) goja.Value {
+		if err := ctx.Err(); err != nil {
+			panic(vm.NewGoError(fmt.Errorf("script cancelled: %w", err)))
+		}
+		callCount++
+		if callCount > e.maxCalls {
+			return vm.ToValue(map[string]any{"ok": false, "error": fmt.Sprintf("exceeded maximum of %d api.call() invocations", e.maxCalls)})
```
Comment (acmacalister) [inline script/engine.go]: The error message says `api.call() invocations` but this is inside `tryCallRendered` — a script that burns through the limit via `callRendered` calls will get a confusing error pointing at the wrong method. The message should say `api calls` (or similar) since the limit is shared across all four call variants.

```go
return vm.ToValue(map[string]any{"ok": false, "error": fmt.Sprintf("exceeded maximum of %d api calls per script", e.maxCalls)})
```

Same copy appears in `tryCall` at line 222 — could update both while here.

File: server/server.go [hunk 2]
```diff
@@ -987,6 +1000,30 @@ func (te *toolExecutor) Execute(ctx context.Context, toolName mcp.ToolName, args
 	return result, err
 }
 
+// ExecuteRendered applies the same markdown/compaction/columnarization pipeline
+// as the execute meta-tool, returning LLM-readable output. Used by api.callRendered().
+func (te *toolExecutor) ExecuteRendered(ctx context.Context, toolName mcp.ToolName, args map[string]any) (*mcp.ToolResult, error) {
+	if toolName == "search" || toolName == "execute" {
```
Comment (acmacalister) [inline server/server.go]: The meta-tool guard is duplicated verbatim between `Execute` and `ExecuteRendered`. If we ever add a third meta-tool or change the error wording we'll need to update it in two places. A small helper would keep this DRY:

```go
func (te *toolExecutor) checkMetaTool(toolName mcp.ToolName) *mcp.ToolResult {
	if toolName == "search" || toolName == "execute" {
		return &mcp.ToolResult{
			Data: fmt.Sprintf(
				"tool %q is a meta-tool and cannot be called from scripts. "+
					"Use the search MCP tool before writing a script to discover tool names.",
				toolName),
			IsError: true,
		}
	}
	return nil
}
```

File: integrations/ollama/types.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+	// context (token ID array) intentionally omitted.
+	timingStats
+}
+
+// --- /api/embed ---
+
+type embedRequest struct {
+	Model      ModelName       `json:"model"`
+	Input      any             `json:"input"`
+	Stream     bool            `json:"stream"`
```
Comment (acmacalister) [inline integrations/ollama/types.go]: `Stream` is always serialized as `"stream":false` in the embed request, but the Ollama `/api/embed` endpoint doesn't support streaming at all — the field is a no-op and just adds noise to the wire payload. Adding `omitempty` (or removing it entirely) would keep the request body clean:

```go
Stream bool `json:"stream,omitempty"`
```

Same note applies to `chatRequest` and `generateRequest`, but for those it's explicitly documenting intent (force non-streaming), so removing it there would reduce clarity.

File: integrations/ollama/compact_specs_test.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+		name := string(toolName)
+		for _, prefix := range mutationPrefixes {
+			assert.NotContains(t, name, "_"+prefix+"_",
+				"mutation tool %q should not have compaction specs", toolName)
+		}
+	}
+}
+
+func TestFieldCompactionSpecs_ExpectedTools(t *testing.T) {
+	expected := []mcp.ToolName{"ollama_list_models", "ollama_list_running"}
```
Comment (acmacalister) [inline integrations/ollama/compact_specs_test.go]: `TestFieldCompactionSpecs_ExpectedTools` only checks that the two list tools have compaction specs, but `ollama_chat`, `ollama_generate`, and `ollama_embed` are also in `rawFieldCompactionSpecs` and worth asserting here. The `NoOrphanSpecs` test already catches specs without handlers, but this test is the only place where we verify the *intent* ("inference tools should strip timing stats"). Adding the three inference tools to `expected` would lock that in.

**Review by acmacalister — APPROVED** [review]
Solid work on both the Ollama integration and the callRendered pipeline. The interface-as-optional-capability pattern for RenderedExecutor is clean, the test coverage is thorough, and the tool descriptions/compaction specs are well-thought-out. Few small things inline.

File: integrations/ollama/types.go [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+	Role      ChatRole        `json:"role"`
+	Content   string          `json:"content"`
+	Thinking  string          `json:"thinking,omitempty"`
+	ToolCalls json.RawMessage `json:"tool_calls,omitempty"`
+}
+
+// --- /api/generate ---
+
+type generateRequest struct {
+	Model     ModelName       `json:"model"`
```
Comment (acmacalister) [inline integrations/ollama/types.go:152]: `Prompt` serializes as `"prompt":""` on the wire whenever the caller doesn't pass a prompt — `generate()` only sets it when non-empty, but the struct tag lacks `omitempty`. `Suffix`, `System`, and all the other optional string fields in this struct consistently use `omitempty`, so this is the odd one out.

```go
Prompt    string          `json:"prompt,omitempty"`
```

Worth adding a test case alongside `TestGenerate` that verifies an empty `prompt` field isn't sent (similar to how `TestEmbed_Success` at line 116 asserts `assert.Nil(t, body["stream"])`).

**Review by acmacalister — APPROVED** [review]
Clean PR — builds, tests pass, lint clean, all three previous review comments addressed. One small thing inline.
